happytransformer/happy_generation.py

```diff
@@ -0,0 +1,128 @@
+"""
+Contains the HappyGeneration class
+"""
+from dataclasses import dataclass
+from transformers import AutoModelForCausalLM
+from happytransformer.happy_transformer import HappyTransformer
+from happytransformer.gen.trainer import GENTrainer, GENTrainArgs, GENEvalArgs
+from happytransformer.adaptors import get_adaptor
+from happytransformer.gen import ARGS_GEN_TRAIN, ARGS_GEN_EVAl, ARGS_GEN_TEST
+from happytransformer.happy_trainer import EvalResult
+from happytransformer.fine_tuning_util import create_args_dataclass
+
+"""
+The main settings that users will adjust when performing experiments
+
+They may still modify all of the settings found here:
+    https://huggingface.co/transformers/main_classes/model.html#generation.
+The values for full_settings are the same as the default values above except for min and max length. 
+"""
+@dataclass
+class GENSettings:
+    do_sample: bool = False
+    early_stopping: bool = False
+    num_beams: int = 1
+    temperature: float = 1
+    top_k: int = 50
+    no_repeat_ngram_size: int = 0
+
+@dataclass
+class GenerationResult:
+    text: str
+
+
+class HappyGeneration(HappyTransformer):
+    """
+    A user facing class for text generation
+    """
+    def __init__(self, model_type: str = "GPT2", model_name: str = "gpt2", load_path: str = ""):
+
+        self.adaptor = get_adaptor(model_type)
+
+        if load_path != "":
+            model = AutoModelForCausalLM.from_pretrained(load_path)
+        else:
+            model = AutoModelForCausalLM.from_pretrained(model_name)
```

Comment:
equivalent to `model = AutoModelForCausalLM.from_pretrained(load_path or model_name)`

Reply:
I find the current way to be more understandable.